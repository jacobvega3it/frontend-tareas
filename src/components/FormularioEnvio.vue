<template>
    <form @submit.prevent="handleSubmit">
        <p><input v-model="email" placeholder="Ingrese un email" /></p>
        <p><button type="submit">Enviar</button></p>
    </form>
</template>

<script>
import { ref } from 'vue';

export default {
    setup() {
        const email = ref('');

        const handleSubmit = () => {
            alert(`Email: ${email.value}`);
            console.log(`Email: ${email.value}`);
        };

        return {
            email,
            handleSubmit
        };
    },
};
</script>

