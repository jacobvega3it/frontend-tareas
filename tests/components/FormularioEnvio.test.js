import FormularioEnvio from "@/components/FormularioEnvio.vue";
import { describe, expect, test } from "vitest";
import { mount } from "@vue/test-utils";

describe("FormularioEnvio", () => {

    test("should match snapshot", () => {
        const wrapper = mount(FormularioEnvio); 
        console.log(wrapper.html());
        expect(wrapper.html()).toMatchSnapshot();
    });

    test("renders form correctly", () => {
        const wrapper = mount(FormularioEnvio); 
        expect(wrapper.find("form").exists()).toBe(true);
        expect(wrapper.find("button").text()).contains("Enviar");
        expect(wrapper.find('input').exists()).toBe(true);
    });

});
