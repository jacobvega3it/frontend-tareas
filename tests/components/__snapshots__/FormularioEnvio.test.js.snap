// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormularioEnvio > should match snapshot 1`] = `
"<form>
  <p><input placeholder="Ingrese un email"></p>
  <p><button type="submit">Enviar</button></p>
</form>"
`;
